src/gurt/telemetry.c

```diff
@@ -2513,6 +2554,224 @@ get_unique_shmem_key(const char *path, int id)
 	return (key_t)d_hash_string_u32(salted, sizeof(salted));
 }
 
+static int
+shm_stat_key(key_t key, struct shmid_ds *shminfo, int *shmid_ptr)
+{
+	int shmid;
+	int rc;
+
+	if (unlikely(shminfo == NULL)) {
+		D_ERROR("NULL shminfo\n");
+		return -DER_INVAL;
+	}
+
+	rc = shmget(key, 0, 0);
+	if (rc < 0) {
+		D_ERROR("failed to get shmid for key 0x%x\n", key);
+		return rc;
+	}
+	shmid = rc;
+
+	rc = shmctl(shmid, IPC_STAT, shminfo);
+	if (rc != 0) {
+		D_ERROR("shmctl(%d, IPC_STAT) failed: %d\n", shmid, rc);
+		return rc;
+	}
+
+	if (shmid_ptr != NULL)
+		*shmid_ptr = shmid;
+
+	return 0;
+}
+
+/*
+ * Set the child segment's ownership to match the parent segment.
+ * Needed in the client telemetry case where the client is allowing
+ * the agent to manage its telemetry segments.
+ */
+static int
+sync_attached_segment_uid(char *path, key_t child_key)
```

Comment:
Shouldn't be necessary to pass the child key. The child key is the value of the link node.

Reply:
This change didn't work... Trying to access the value of the link node here resulted in a segfault.